File: solutions/Best-Time-to-Buy-and-Sell-Stock-II/Best-Time-to-Buy-and-Sell-Stock-II.py


class Solution:
    # @param prices, a list of integer
    # @return an integer
    def maxProfit(self, prices):
        last=10000
        ans=0
        for i in range(len(prices)):
            if prices[i]>last:
                ans+=prices[i]-last
            last=prices[i]
        return ans
 

File: solutions/Best-Time-to-Buy-and-Sell-Stock-II/Best-Time-to-Buy-and-Sell-Stock-II.cpp

// class Solution {
// public:
//     int maxProfit(vector<int> &prices) {
//         int ans=0;
//         for(int i=0;i<prices.size();i++)
//             if(i>0 && prices[i]>prices[i-1])
//                 ans+=prices[i]-prices[i-1];
//         return ans;
//     }
// };

// 2020-04-06
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        for(int i=0;i<prices.size();i++) {
            if(i > 0 && prices[i] > prices[i-1]) {
                ans += (prices[i] - prices[i-1]);
            }
        }
        return ans;
    }
};
 